dev-support/test-patch.d/scaladoc.sh

```diff
@@ -0,0 +1,139 @@
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+add_plugin scaladoc
+
+
+function scaladoc_filefilter
+{
+  local filename=$1
+
+  if [[ ${filename} =~ \.scala$ ]]; then
+   yetus_debug "tests/scaladoc: ${filename}"
+   add_test scaladoc
+  fi
+}
+
+## @description  Confirm Javadoc pre-patch
+## @audience     private
+## @stability    stable
+## @replaceable  no
+## @return       0 on success
+## @return       1 on failure
+function scaladoc_preapply
+{
+  local result=0
+
+  verify_needed_test scaladoc
+  if [[ $? == 0 ]]; then
+     return 0
+  fi
+
+  big_console_header "Pre-patch ${PATCH_BRANCH} Javadoc verification"
+
+  personality_modules branch scaladoc
+  ${BUILDTOOL}_modules_worker branch scaladoc
+
+  ((result=result + $?))
+  modules_messages branch scaladoc true
+
+
+  if [[ ${result} -gt 0 ]]; then
+    return 1
+  fi
+  return 0
+}
+
+## @description  Count and compare the number of JavaDoc warnings pre- and post- patch
+## @audience     private
+## @stability    evolving
+## @replaceable  no
+## @return       0 on success
+## @return       1 on failure
+function scaladoc_postinstall
+{
+  local i
+  local result=0
+  local fn
+  declare -i numbranch=0
+  declare -i numpatch=0
+
+  verify_needed_test scaladoc
+    if [[ $? == 0 ]]; then
+    return 0
+  fi
+
+  big_console_header "Determining number of patched scaladoc warnings"
+
+  personality_modules patch scaladoc
+  ${BUILDTOOL}_modules_workers patch scaladoc
+
+  i=0
+  until [[ ${i} -eq ${#MODULE[@]} ]]; do
+    if [[ ${MODULE_STATUS[${i}]} == -1 ]]; then
+      ((result=result+1))
+      ((i=i+1))
+      continue
+    fi
+
+    fn=$(module_file_fragment "${MODULE[${i}]}")
+    module_suffix=$(basename "${MODULE[${i}]}")
+    if [[ ${module_suffix} == \. ]]; then
+      module_suffix=root
+    fi
+
+    if [[ -f "${PATCH_DIR}/branch-scaladoc-${fn}.txt" ]]; then
+      ${GREP} -i scaladoc "${PATCH_DIR}/branch-scaladoc-${fn}.txt" \
+        > "${PATCH_DIR}/branch-scaladoc-${fn}-warning.txt"
+    else
+      touch "${PATCH_DIR}/branch-scaladoc-${fn}.txt" \
+        "${PATCH_DIR}/branch-scaladoc-${fn}-warning.txt"
+    fi
+
+    if [[ -f "${PATCH_DIR}/patch-scaladoc-${fn}.txt" ]]; then
+      ${GREP} -i scaladoc "${PATCH_DIR}/patch-scaladoc-${fn}.txt" \
+        > "${PATCH_DIR}/patch-scaladoc-${fn}-warning.txt"
+    else
+      touch "${PATCH_DIR}/patch-scaladoc-${fn}.txt" \
+        "${PATCH_DIR}/patch-scaladoc-${fn}-warning.txt"
+    fi
```

Comment:
shellcheck: 17: note: Double quote to prevent globbing and word splitting. [SC2086]